bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/modbus/ISolarmanProtocol.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarman.internal.modbus;
+
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.solarman.internal.modbus.exception.SolarmanException;
+
+/**
+ * @author Peter Kretz - Initial contribution
+ */
+@NonNullByDefault
+public interface ISolarmanProtocol {
```

Comment:
nit: I think that prefixing interfaces with I is an anti-pattern.

Reply:
I will change this